apps/emqx_durable_storage/src/emqx_ds_beamformer_rt.erl

```diff
@@ -101,46 +120,42 @@ init([CBM, DBShard, Name, _Opts]) ->
         sub_tab = emqx_ds_beamformer:make_subtab(DBShard),
         metrics_id = emqx_ds_beamformer:metrics_id(DBShard, rt),
         name = Name,
-        high_watermark = ets:new(high_watermark, [ordered_set, private]),
+        high_watermark = ets:new(high_watermark, [ordered_set, public]),
         queue = emqx_ds_beamformer_waitq:new(),
         batch_size = emqx_ds_beamformer:cfg_batch_size()
     },
     self() ! ?housekeeping_loop,
     {ok, S}.
 
-handle_call(Req = #sub_state{}, _From, S) ->
-    Reply = do_enqueue(Req, S),
-    {reply, Reply, S};
+handle_call(Req = #sub_state{}, From, S0) ->
+    Fun = fun(S) ->
+        do_enqueue(Req, From, S)
+    end,
+    push_cmd(S0, Fun);
 handle_call(_Call, _From, S) ->
     {reply, {error, unknown_call}, S}.
 
-handle_cast(#seal_event{rank = Rank}, S = #s{queue = Queue}) ->
-    %% Currently generation seal events are treated as stream events,
-    %% for each known stream that has the matching rank:
-    Streams = emqx_ds_beamformer_waitq:streams_of_rank(Rank, Queue),
-    _ = [process_stream_event(false, Stream, S) || Stream <- Streams],
-    {noreply, S};
+handle_cast(#seal_event{rank = Rank}, S0) ->
+    Fun = fun(S = #s{queue = Queue}) ->
+        %% Currently generation seal events are treated as stream events,
+        %% for each known stream that has the matching rank:
+        Streams = emqx_ds_beamformer_waitq:streams_of_rank(Rank, Queue),
+        %% TODO: pass it through the event queue too
+        _ = [process_stream_event(self(), false, Stream, S) || Stream <- Streams],
```

Comment:
Q: That would be pid of a worker, not the beamformer process, wondering if this is intentional.